Submitted By: Jim Gifford (jim at cross-lfs dot org)
Date: 08-22-2009
Initial Package Version: 2.10.1
Origin: Upstream
Upstream Status: Applied
Description: These are fixes eglibc-2.10.1, and should be
             rechecked periodically.

diff -Naur libc.orig/elf/ldd.bash.in libc/elf/ldd.bash.in
--- libc.orig/elf/ldd.bash.in	2009-02-13 10:31:14.000000000 -0800
+++ libc/elf/ldd.bash.in	2009-08-22 12:59:18.000000000 -0700
@@ -1,4 +1,4 @@
-#! @BASH@
+#! /bin/bash
 # Copyright (C) 1996-2008, 2009 Free Software Foundation, Inc.
 # This file is part of the GNU C Library.
 
diff -Naur libc.orig/libio/Makefile libc/libio/Makefile
--- libc.orig/libio/Makefile	2009-07-02 13:03:14.000000000 -0700
+++ libc/libio/Makefile	2009-08-22 12:59:18.000000000 -0700
@@ -170,6 +170,7 @@
 tst-fopenloc-ENV = LOCPATH=$(common-objpfx)localedata \
 		   MALLOC_TRACE=$(objpfx)tst-fopenloc.mtrace
 tst-fgetws-ENV = LOCPATH=$(common-objpfx)localedata
+tst-fgetwc-ENV = LOCPATH=$(common-objpfx)localedata
 tst-ungetwc1-ENV = LOCPATH=$(common-objpfx)localedata
 tst-ungetwc2-ENV = LOCPATH=$(common-objpfx)localedata
 bug-ungetwc2-ENV = LOCPATH=$(common-objpfx)localedata
diff -Naur libc.orig/sunrpc/Makefile libc/sunrpc/Makefile
--- libc.orig/sunrpc/Makefile	2008-04-25 04:12:05.000000000 -0700
+++ libc/sunrpc/Makefile	2009-08-22 12:59:18.000000000 -0700
@@ -144,11 +144,11 @@
 	$(+link)
 
 $(addprefix $(objpfx)cross-,$(rpcgen-objs)): $(objpfx)cross-%.o: %.c
-	gcc $< -c -D_RPC_THREAD_SAFE_ -D_CROSS_RPCGEN_ \
+	$(BUILD_CC) $< -c -D_RPC_THREAD_SAFE_ -D_CROSS_RPCGEN_ \
 		$(OUTPUT_OPTION) $(compile-mkdep-flags)
 
 $(objpfx)cross-rpcgen: $(addprefix $(objpfx)cross-,$(rpcgen-objs))
-	gcc $^ -o $@
+	$(BUILD_CC) $^ -o $@
 
 # This makes sure -DNOT_IN_libc is passed for all these modules.
 cpp-srcs-left := $(rpcgen-objs:.o=.c)
diff -Naur libc.orig/timezone/Makefile libc/timezone/Makefile
--- libc.orig/timezone/Makefile	2009-03-03 05:14:09.000000000 -0800
+++ libc/timezone/Makefile	2009-08-22 12:59:18.000000000 -0700
@@ -181,11 +181,11 @@
 $(objpfx)zic: $(addprefix $(objpfx), $(zic-objs))
 
 $(addprefix $(objpfx)cross-,$(zic-objs)): $(objpfx)cross-%.o: %.c
-	gcc $< -c $(OUTPUT_OPTION) $(CFLAGS-$*.c) $(CPPFLAGS-$*) \
+	$(BUILD_CC) $< -c $(OUTPUT_OPTION) $(CFLAGS-$*.c) $(CPPFLAGS-$*) \
 		-DCROSS_ZIC $(compile-mkdep-flags)
 
 $(objpfx)cross-zic: $(addprefix $(objpfx)cross-,$(zic-objs))
-	gcc $(addprefix $(objpfx)cross-,$(zic-objs)) -o $@
+	$(BUILD_CC) $(addprefix $(objpfx)cross-,$(zic-objs)) -o $@
 
 tz-cflags = -DTZDIR='"$(zonedir)"' \
 	    -DTZDEFAULT='"$(localtime-file)"' \
